modules/creating-manifest-file-multiple-interfaces-br-ex.adoc

```diff
@@ -0,0 +1,187 @@
+[source,ini]
+----
+[connection]
+id=bond1
+type=bond
+interface-name=bond1
+autoconnect=true
+connection.autoconnect-slaves=1
+autoconnect-priority=20
+
+[bond]
+mode=802.3ad
+miimon=100
+xmit_hash_policy="layer3+4"
+
+[ipv4]
+method=auto
+----
++
+.. Convert the definition files to Base64 encoded strings by running the following command:
++
+[source,terminal]
+----
+$ base64 <directory_path>/en01.config
+----
++
+[source,terminal]
+----
+$ base64 <directory_path>/eno2.config
+----
++
+[source,terminal]
+----
+$ base64 <directory_path>/bond1.config
+----
+
+. Prepare the environment variables. Replace `<machine_role>` with the node role, such as `worker`, and replace `<interface_name>` with the name of your additional `br-ex` bridge name.
++
+[source,terminal]
+----
+$ export ROLE=<machine_role>
+----
++
+[source,terminal]
+----
+$ export INTERFACE=<interface_name>
+----
+
+. Create a machine config manifest file for configuring the network plugin by entering the following command in your terminal:
++
+[source,terminal]
+----
+$ oc create -f <machine_config_file_name>
+----
+
+. Create an Open vSwitch (OVS) bridge, `br-ex1`, on nodes by using the OVN-Kubernetes network plugin to create an `extra_bridge` file`. Ensure that you save the file in the `/etc/ovnk/extra_bridge` path of the host. The file must state the interface name that supports the additional bridge and not the default interface that supports `br-ex`, which holds the primary IP address of the node. 
++
+.Example configuration for the `extra_bridge` file, `/etc/ovnk/extra_bridge`, that references a secondary interface that is attached to a VLAN
+[source,text]
+----
+bond1
+----
+
+. Define each interface definition in a machine config manifest file:
++
+.Example of a machine config file with definitions added for `bond1`, `eno1`, and `en02`
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 12-${ROLE}-sec-bridge-cni
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+      - contents:
+          source: data:;base64,<base-64-encoded-contents-for-bond1.conf>
+        path: /etc/NetworkManager/system-connections/bond1.nmconnection
+        filesystem: root
+        mode: 0600
+      - contents:
+          source: data:;base64,<base-64-encoded-contents-for-eno1.conf>
+        path: /etc/NetworkManager/system-connections/eno1.nmconnection
+        filesystem: root
+        mode: 0600
+      - contents:
+          source: data:;base64,<base-64-encoded-contents-for-eno2.conf>
+        path: /etc/NetworkManager/system-connections/eno2.nmconnection
+        filesystem: root
+        mode: 0600
+# ...
+----
+
+. Bind `br-ex1` to the secondary interface by configuring the `/var/lib/ovnk/iface_default_hint` file. With this file, you can change the priority of what bridge gets selected for your nodes.  
```

Comment:
> Bind `br-ex1` to the secondary interface by configuring the `/var/lib/ovnk/iface_default_hint` file. With this file, you can change the priority of what bridge gets selected for your nodes.

When we create /etc/ovnk/extra_bridge, this file specifies the interface to be used by `br-ex1` and is it's own procedure. 

The iface_default_hint file specifies the interface to be used by `br-ex`

suggested copy change: (I would make a new header for this entirely:)

~~~
Optional: Overriding the default interface selection logic for br-ex

You can override the default selection method that OVN-kubernetes will use to select an interface on node boot and ensure that br-ex selects a newly created bond interface instead of one of the primary interfaces that hosts the IP of the node.

This can be accomplished by creating a machine-configuration update that defines the file: `/var/lib/ovnk/iface_default_hint`. This file defines the interface that you would like `br-ex` to bind to and use for node traffic (for example, a bond or vlan that is defined as a day-two operation).

The following procedure shows how to define this file:

.Procedure

. Create a machine-configuration that defines the existing static interface that will be used to host br-ex on subsequent node restarts. For example, the below configuration defines `bond0` as the interface that should host br-ex:

[source,yaml]
----
apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  labels:
    machineconfiguration.openshift.io/role: worker
  name: 12-worker-default-bridge
spec:
  config:
    ignition:
      version: 3.2.0
    storage:
      files:
        - path: /var/lib/ovnk/iface_default_hint
          mode: 0420
          overwrite: true
          contents:
            source: data:text/plain;charset=utf-8,bond0
          filesystem: root
----

. Apply the machine-configuration to your selected nodes:

[source,terminal]
---
oc create -f <machine_config_file_name>
---
~~~

//additional:
You can remove lines 163 --> 168:

~~~
.Example `iface_default_hint` file that states an additional bridge named `ens2f0.4085`
[source,text]
----
DEFAULT=<br-ex-interface_name> 
BR2=<br-ex1-interface_name> <1>
----
<1> Based on the examples in the procedure, `<br-ex1-interface_name>` translates to `bond1`.
~~~